Python: Getting Started grpc (#7816)
getsentry/sentry-docs: src/platforms/python/common/configuration/integrations/grpc.mdx

Add appropriate interceptor to your grpc stub/server:

<SignInNote />

### Server

```python
import grpc

import sentry_sdk
from sentry_sdk.integrations.grpc.server import ServerInterceptor

sentry_sdk.init(
    dsn="___PUBLIC_DSN___",
    # Set traces_sample_rate to 1.0 to capture 100%
    # of transactions for performance monitoring.
    traces_sample_rate=1.0,
)

...

server = grpc.server(
    thread_pool=...,
    interceptors=[ServerInterceptor()],
)
```

### Client

```python
import grpc

import sentry_sdk
from sentry_sdk.integrations.grpc.client import ClientInterceptor

sentry_sdk.init(
    dsn="___PUBLIC_DSN___",
    # Set traces_sample_rate to 1.0 to capture 100%
    # of transactions for performance monitoring.
    traces_sample_rate=1.0,
)

...

with grpc.insecure_channel("example.com:12345") as channel:
    channel = grpc.intercept_channel(channel, *[ClientInterceptor()])
```

## Verify

### Server

> If you added the interceptor like described above, the server will create a transaction that will show up in Sentry.
> 
> It takes a couple moments until the data shows up in Sentry.

Reviewer: ```suggestion
If you added the interceptor as described above, the server will create a transaction that will show up in [sentry.io](https://sentry.io).

It takes a couple of moments for the data to appear in [sentry.io](https://sentry.io).
```

### Client

```python
import grpc

sentry_sdk.init(...)  # same as above

...

with sentry_sdk.start_transaction(op="function", name="grcp_client"):
    with grpc.insecure_channel("example.com:12345") as channel:
        channel = grpc.intercept_channel(channel, *[ClientInterceptor()])
        stub = helloworld_pb2_grpc.GreeterStub(channel)
        response = stub.SayHello(helloworld_pb2.HelloRequest(name="you"))

```

> This will create a transaction called `testing_sentry` in the Performance section of Sentry and will create a span for the call to the `SayHello` method on the server.
> 
> It takes a couple moments until the data shows up in Sentry.

Reviewer: ```suggestion
This will create a transaction called `testing_sentry` in the Performance section of [sentry.io](https://sentry.io) and will create a span for the call to the `SayHello` method on the server.

It takes a couple of moments for the data to appear in [sentry.io](https://sentry.io).
```
